src/signing/env_mnemonic.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+use crate::account::Account;
+
+use std::{collections::HashMap, env};
+
+use bech32::ToBase32;
+use bee_common::packable::Packable;
+use blake2::{
+    digest::{Update, VariableOutput},
+    VarBlake2b,
+};
+use hmac::Hmac;
+use iota::{Ed25519Signature, ReferenceUnlock, SignatureUnlock, UnlockBlock};
+use rand::{thread_rng, Rng};
+use unicode_normalization::UnicodeNormalization;
+
+use bee_signing_ext::{
+    binary::{ed25519, BIP32Path},
+    Signer,
+};
+
+const PBKDF2_ROUNDS: usize = 2048;
+const PBKDF2_BYTES: usize = 32; // 64 for secp256k1 , 32 for ed25
+
+/// PBKDF2 helper, used to generate [`Seed`][Seed] from [`Mnemonic`][Mnemonic]
+///
+/// [Mnemonic]: ../mnemonic/struct.Mnemonic.html
+/// [Seed]: ../seed/struct.Seed.html
+fn _pbkdf2(input: &[u8], salt: &str) -> Vec<u8> {
+    let mut seed = vec![0u8; PBKDF2_BYTES];
+    pbkdf2::pbkdf2::<Hmac<sha2::Sha512>>(input, salt.as_bytes(), PBKDF2_ROUNDS, &mut seed);
+    seed
+}
+
+fn mnemonic_to_ed25_seed(mnemonic: String, password: String) -> ed25519::Ed25519Seed {
+    let salt = format!("mnemonic{}", password);
+    let normalized_salt = salt.nfkd().to_string();
+    let bytes = _pbkdf2(mnemonic.as_bytes(), &normalized_salt);
+    ed25519::Ed25519Seed::from_bytes(&bytes).unwrap()
+}
+
+fn derive_into_address(private_key: ed25519::Ed25519PrivateKey) -> String {
+    let public_key = private_key.generate_public_key().to_bytes();
+    // Hash the public key to get the address
+    let mut hasher = VarBlake2b::new(32).unwrap();
+    hasher.update(public_key);
+    let mut result = vec![];
+    hasher.finalize_variable(|res| {
+        result = res.to_vec();
+    });
+
+    let prefix: Vec<u8> = vec![1]; // prefix for ed25 addresses
+    let mut data = prefix;
+    data.append(&mut result);
+    bech32::encode("iota", data.to_base32()).unwrap()
+}
+
+#[derive(Default)]
+pub struct EnvMnemonicSigner;
+
+impl EnvMnemonicSigner {
+    fn get_seed(&self) -> ed25519::Ed25519Seed {
+        mnemonic_to_ed25_seed(
+            env::var("IOTA_WALLET_MNEMONIC")
```

Comment:
Having the mnemonic in the env looks weird tbh, out of curiosity, what's the use-case ?

Reply:
maybe @cvarley100 or @laumair can expand on this topic